QMigrate/signal/scan.py

```diff
@@ -485,29 +505,47 @@ def _read_event_waveform_data(self, w_beg, w_end):
 
         # Extra pre- and post-pad default to 0.
         pre_pad = post_pad = 0.
-        if self.pre_cut:
+
+        if self.pre_cut or self.calc_magnitudes:
+            if self.calc_magnitudes and self.pre_cut:
+                pre_cut = max(self.mags.amp.noise_window + self.marginal_window,
+                              self.pre_cut)
+            elif self.calc_magnitudes:
+                pre_cut = self.mags.amp.noise_window + self.marginal_window
+            else:
+                pre_cut = self.pre_cut
             # only subtract 1*marginal_window so if the event otime moves by
             # this much the selected pre_cut can still be applied
-            pre_pad = self.pre_cut - self.marginal_window - self.pre_pad
-            if pre_pad < 0:
+            pre_pad = pre_cut - self.marginal_window - self.pre_pad
+            if pre_pad < 0 and self.pre_cut:
                 msg = (f"\t\tWarning: specified pre_cut {self.pre_cut} is"
                        "shorter than default pre_pad\n"
                        f"\t\t\tCutting from pre_pad = {self.pre_pad}")
                 logging.info(msg)
                 pre_pad = 0.
 
-        if self.post_cut:
+        if self.post_cut or self.calc_magnitudes:
+            if self.calc_magnitudes and self.post_cut:
+                post_cut = max(((1 + self.picker.fraction_tt) *
+                                self.lut.max_ttime + self.marginal_window +
+                                self.mags.amp.signal_window), self.post_cut)
+            elif self.calc_magnitudes:
+                post_cut = ((1 + self.picker.fraction_tt) *
+                            self.lut.max_ttime + self.marginal_window +
+                            self.mags.amp.signal_window)
+            else:
+                post_cut = self.post_cut
             # only subtract 1*marginal_window so if the event otime moves by
             # this much the selected post_cut can still be applied
-            post_pad = self.post_cut - self.marginal_window - self.post_pad
-            if post_pad < 0:
+            post_pad = post_cut - self.marginal_window - self.post_pad
+            if post_pad < 0 and self.post_cut:
                 msg = (f"\t\tWarning: specified post_cut {self.post_cut} is"
                        " shorter than default post_pad\n"
                        f"t\t\tCutting to post_pad = {self.post_pad}")
                 logging.info(msg)
                 post_pad = 0.
```

Comment:
Latest possible end time of the "signal window" for amplitude measurement is that. The default window definition for amplitude measurement is currently based heavily off that used for the picker. Rather than re-specifying the fraction_tt, I pull it off the picker.

This can be handled better. Medium term I want to review the window specification. Additionally to that, I think fraction_tt should perhaps be an attribute of the LUT. The option to override it could still be made available for picker / magnitudes etc.